src/detect-http-headers-stub.h

```diff
@@ -324,30 +128,55 @@ static int DetectHttpHeadersSetup(DetectEngineCtx *de_ctx, Signature *s, const c
     return 0;
 }
 
+/**
+ * \brief this function setup the http.header keyword used in the rule
+ *
+ * \param de_ctx   Pointer to the Detection Engine Context
+ * \param s        Pointer to the Signature to which the current keyword belongs
+ * \param str      Should hold an empty string always
+ *
+ * \retval 0       On success
+ */
+static int DetectHttpHeadersSetupSticky(DetectEngineCtx *de_ctx, Signature *s, const char *str)
+{
+    DetectBufferSetActiveList(s, g_buffer_id);
+    s->alproto = ALPROTO_HTTP;
+    return 0;
+}
+
 static void DetectHttpHeadersRegisterStub(void)
 {
+    sigmatch_table[KEYWORD_ID].name = KEYWORD_NAME_LEGACY;
```

Comment:
this is overwritten by the next sets as all are setting to sigmatch_table[KEYWORD_ID]. The KEYWORD_NAME_LEGACY should be set in sigmatch_table[KEYWORD_ID].alias instead.

Reply:
Should there be two `sigmatch_table` entries corresponding `http.name` and `http_name` as is done here: https://github.com/OISF/suricata/blob/master/src/detect-http-cookie.c#L86:L103?

Or, should the `.alias` member be set to the legacy value?

Reply:
The reason why the cookie case is different is that there we went from the content modifier
```
content:"abc"; http_cookie;
```
to the sticky buffer:
```
http.cookie; content:"abc";
```
These have very different mechanics during the rule parsing, so they are registered as 2 different keywords.

In this case of the headers all keywords are already sticky buffers, so we can simply register the new name and keep the old name as an alias.

Reply:
Understood. Will fix.